server/subscribe.go

```diff
@@ -170,6 +171,12 @@ func (p *Plugin) matchesSubsciptionFilters(wh *webhook, filters SubscriptionFilt
 		}
 
 		value := getIssueFieldValue(issue, field.Key)
+		if visibilityAttribute != "" {
+			value[visibilityAttribute] = true
+		} else if field.Key == CommentVisibility {
+			value[visibleToAllUsers] = true
+		}
+
```

Comment:
I think this may read a little better, as the implementation of `StringSet` as a map should be hidden away:

```suggestion
		value := getIssueFieldValue(issue, field.Key)
		if visibilityAttribute != "" {
			value.Add(visibilityAttribute)
		} else if field.Key == CommentVisibility {
			value.Add(visibleToAllUsers)
		}
```

Also I think the first block be inside a `field.Key == CommentVisibility` condition like so:

```go
		value := getIssueFieldValue(issue, field.Key)
		if field.Key == CommentVisibility {
			if visibilityAttribute != "" {
				value.Add(visibilityAttribute)
			} else {
				value.Add(visibleToAllUsers)
			}
		}
```